collect_stata/write_json.py

```diff
@@ -149,105 +166,107 @@ def stats_number(elem, file_csv):
     }
 
 
-def uni_statistics(elem, file_csv):
-    """Call function to generate statistics depending on the variable type
+def uni_statistics(elem, data):
+    """Call function to generate statistics depending on the variable type.
 
-    Input:
-    elem: dict
-    file_csv: pandas DataFrame
+    Args:
+        elem (dict): Contains information of one variable.
+        data (pandas.DataFrame): Datatable of imported data.
 
-    Output:
-    statistics: OrderedDict
+    Returns:
+        statistics (OrderedDict):
+        Statistics for either categorical, nominal or numerical variables.
     """
 
     if elem["type"] == "cat":
 
-        statistics = stats_cat(elem, file_csv)
+        statistics = stats_cat(elem, data)
 
     elif elem["type"] == "string":
 
-        statistics = stats_string(elem, file_csv)
+        statistics = stats_string(elem, data)
 
     elif elem["type"] == "number":
 
-        statistics = stats_number(elem, file_csv)
+        statistics = stats_number(elem, data)
 
     else:
         statistics = dict()
 
     return statistics
 
 
-def uni(elem, file_csv):
-    """Call function to generate frequencies depending on the variable type
+def uni(elem, data):
+    """Call function to generate frequencies depending on the variable type.
 
-    Input:
-    elem: dict
-    file_csv: pandas DataFrame
+    Args:
+        elem (dict): Contains information of one variable.
+        data (pandas.DataFrame): Datatable of imported data.
 
-    Output:
-    statistics: OrderedDict
+    Returns:
+        statistics (OrderedDict):
+        Statistics for either categorical, nominal or numerical variables.
     """
 
     statistics = OrderedDict()
     _type = elem["type"]
     type_functions = {"string": uni_string, "number": uni_number}
 
     if _type == "cat":
-        statistics.update(uni_cat(elem, file_csv))
+        statistics.update(uni_cat(elem, data))
     # We change this to else, if no other types exist
     elif _type in type_functions:
         statistics.update(type_functions[_type]())
 
     return statistics
 
 
-def stat_dict(elem, file_csv, file_json, study):
+def stat_dict(elem, data, metadata, study: str):
     """Fill variables with metadata of the dataset.
 
-    Input:
-    elem: dict
-    file_csv: pandas DataFrame
-    file_json: dict
-    study: string
+    Args:
+        elem (dict): Contains information of one variable.
+        data (pandas.DataFrame): Datatable of imported data.
+        metadata (dict): Metadata of the imported data.
+        study (str): Name of the study.
 
-    Output:
-    meta_dict: OrderedDict
+    Returns:
+        meta_dict (OrderedDict): Combine calculations and meta information.
     """
 
     scale = elem["type"][0:3]
 
     meta_dict = OrderedDict()
```

Comment:
Could you include the following lines into the initialization of the dict?

Reply:
I also do not understand what you mean.

Reply:
Maybe the highlighting from github was not ideal.
I meant to iniitalize the dictionary with the values in one "go".
```python
meta_dict = {
    "study": study
    ...
}
```

or

```python
meta_dict = dict(
   study=study
    ...
)
```